cli/flags.go

```diff
@@ -685,6 +686,31 @@ func (f *FlagSection) Uint64Var(i *Uint64Var) {
 	})
 }
 
+// LoggerVar is a special flag that binds to a [logging.Logger].
+func (f *FlagSection) LoggerVar(l *logging.Logger) {
```

Comment:
I like this design better, but I don't love the devex. I think this is what I really want:

```go
type MyCommand struct {
  logger *logging.Logger
}

func (c *MyCommand) Flags() {
  set := c.NewFlagSet()
  set.AddLoggingFlags(&c.logger)
  return set
}
```

And that would automatically create a section and add multiple flags:

```go
func (s *FlagSet) AddLoggingFlags(l *logging.Logger) {
  f := s.NewSection("LOGGING OPTIONS")

  f.StringVar(&StringVar{
    Name: "log-level",
    EnvVar: "LOG_LEVEL",
    Default: logging.LevelWarn,
    Target: &l.Level,
  })

  f.StringVar(&StringVar{
    Name: "log-format",
    EnvVar: "LOG_FORMAT",
    Default: logging.FormatJSON,
    Target: &l.Format,
  })
}
```

We might end up needing custom getters/setters as you've done here, but I think this is the API we want. WDYT?

Reply:
I don't think I can get this to work without refactoring the logging package. I will close this PR and do the logging refactoring first.